src/cpp/include/openvino/genai/vlm_pipeline.hpp

```diff
@@ -0,0 +1,173 @@
+    // history between generate() calls.
+    bool m_is_chat_conversation;
+    ChatHistory m_history;
+    std::string m_templated_chat_history;
+    size_t image_id = 0;  // Used to insert <image_id>i</image_id> per image (not a slice).
+
+    /// @brief Construct a pipeline form a folder containing tokenizer
+    /// and model IRs.
+    /// @param model_dir A folder to read tokenizer and model IRs.
+    /// @param device Inference device. A tokenizer is always compiled
+    /// for CPU.
+    /// @param device_config A config to pass to ov::Core.set_property()
+    /// and ov::Core::compile_model().
+    /// @param core ov::Core instance to use.
+    explicit VLMPipeline(
+        const std::filesystem::path& model_dir,
+        const std::string& device="CPU",
+        const ov::AnyMap device_config={},
+        ov::Core core=ov::Core{}
+    ) : VLMPipeline{
+        model_dir,
+        Tokenizer(model_dir.string(), device_config),
+        device,
+        device_config,
+        core
+    } {}
+
+    /// @brief Construct a pipeline form a folder containing model IRs
+    /// and from a Tokenizer instance.
+    /// @param model_dir A folder to read model IRs.
+    /// @param tokenizer An instance of Tokenizer to use.
+    /// @param device Inference device.
+    /// @param device_config A config to pass to ov::Core.set_property()
+    /// and ov::Core::compile_model().
+    /// @param core ov::Core instance to use.
+    VLMPipeline(
+        const std::filesystem::path& model_dir,
+        const ov::genai::Tokenizer& tokenizer,
+        const std::string& device="CPU",
+        const ov::AnyMap device_config={},
+        ov::Core core=ov::Core{}
+    );
+
+    /// @brief Default destructor.
+    ~VLMPipeline();
+
+    /// @brief Generate a response given a prompt and any number of
+    /// uint8 RGB images.
+    /// @param prompt A prompt to respond to.
+    /// @param images Images to be prepended to a prompt.
+    /// @param generation_config A config to follow for text generation.
+    /// @param streamer A streamer to acquire intermidiate result.
+    /// @return A string generated by a model.
+    DecodedResults generate(
+        const std::string& prompt,
+        const std::vector<ov::Tensor>& rgbs,
+        const GenerationConfig& generation_config,
+        const StreamerVariant& streamer
+    );
+    /// @brief Generate a response given a prompt and config.
+    /// @param prompt A prompt to respond to.
+    /// @param config_map A config may contain GenerationConfig, values
+    /// for its members, StreamerVariant a single image or multiple
+    /// images.
+    /// @return A string generated by a model.
+    DecodedResults generate(
+        const std::string& prompt,
+        const ov::AnyMap& config_map
+    );
+    /// @brief Generate a response given a prompt and arbitrary number
+    /// of ov::Property instances.
+    /// Example:
+    /// generate("text", image(std::move(rgb)), do_sample(true));
```

Comment:
do we need `std::move` ? looks like noisy not important moment as ov::Tensor is shared ptr by its implementation

Reply:
resolved in https://github.com/Wovchena/openvino.genai-public/pull/68